yaptide/celery/utils/manage_tasks.py

```diff
@@ -87,3 +92,54 @@ def translate_celery_state_naming(job_state: str) -> str:
         return EntityState.COMPLETED.value
     # Others are the same
     return job_state
+
+
+def handle_shieldhit_cancellation(tasks: list[CeleryTaskModel]):
+    """Function cancelling shieldhit processes"""
+    command_as_list = ['kill', '--signal', 'SIGINT']
+    celery_ids = []
+    for task in tasks:
+        if task.task_state == EntityState.RUNNING.value:
+            command_as_list.append(str(task.sim_pid))
+        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value) or task.sim_pid is None:
+            celery_ids.append(task.celery_id)
+
+    # Send SIGINT to Shieldhit processes
+    subprocess.run(command_as_list, check=True)
+    celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
+
+
+def handle_fluka_cancellation(tasks: list[CeleryTaskModel]):
+    """Function cancelling fluka processes"""
+    FILE_NAME = 'rfluka.stop'
+    celery_ids = []
+    for task in tasks:
+        if task.task_state == EntityState.RUNNING.value:
+            try:
+                if (target_folder := next(Path(task.path_to_sim).glob("fluka_*/"), None)):
+                    (target_folder / FILE_NAME).write_text("")
+            except OSError as e:
+                logging.warning("Error due to file operation: %s", str(e))
+        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value) or task.path_to_sim is None:
+            celery_ids.append(task.celery_id)
+    celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
+
+
+def cancel_tasks_without_fetching(simulation: CelerySimulationModel, tasks: list[CeleryTaskModel]):
+    """Function to cancel tasks without feching data"""
+    celery_ids = [
+        task.celery_id for task in tasks
+        if task.task_state in [EntityState.PENDING.value, EntityState.RUNNING.value, EntityState.UNKNOWN.value]
+    ]
+
+    # Revoke the merge task first
+    celery_app.control.revoke(simulation.merge_id, terminate=True, signal="SIGINT")
+    celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
```

Comment:
is that blocking or non-blocking operation ?

Reply:
i.e. after calling `revoke` are you 100% sure that all fluka processes are dead ? or you need to wait till its done ?

Reply:
non-blocking

Reply:
then I doubt if you should immediately set the simulation state to CANCELED if you are not sure that it has already been killed. Killing may take some time. As for now its OK, but I think this should be done in better way later.